src/Raven.Client/ServerWide/DatabaseRecord.cs

```diff
@@ -574,4 +565,27 @@ public DynamicJsonValue ToJson()
             };
         }
     }
+
+    public class ShardingRecord 
+    {
+        public DatabaseTopology[] Shards;
+
+        public List<ShardBucketRange> ShardBucketRanges = new List<ShardBucketRange>();
+
+        /// <summary>
+        /// This is used to customize the distribution of data for specific shards on a particular document id prefix
+        /// </summary>
+        public Dictionary<string, List<ShardBucketRange>> Prefixed;
```

Comment:
need to make sure this is case insensitive

Reply:
We can also add `Dictionary<Range, int> StickyBuckets` which will map from buckets to shard.
But maybe it is for a different PR :-)

Reply:
We never do a lookup here. We only do a scan.
And when we scan, we do that with case insensitive compare.